### PR TITLE
Regenerated CLI using swagger api version 2021-05-01

### PR DESCRIPTION
**Pull request by KalpeshChavan12**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
maintenance

File: src/maintenance/azext_maintenance/azext_metadata.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "azext.isPreview": true,
-    "azext.minCliCoreVersion": "2.3.1"
+    "azext.isExperimental": true,
```
Comment (kairu-ms) [inline src/maintenance/azext_maintenance/azext_metadata.json:2]: You cannot change it from isPreview to isExperimental.

File: src/maintenance/azext_maintenance/generated/_params.py [hunk 2]
```diff
@@ -23,188 +23,219 @@
...
-        c.argument('provider_name', type=str, help='Resource provider name')
-        c.argument('resource_type', type=str, help='Resource type')
-        c.argument('resource_name', type=str, help='Resource identifier')
-        c.argument('apply_update_name', options_list=['--name', '-n', '--apply-update-name'], type=str, help=''
-                   'applyUpdate Id')
+        c.argument('provider_name', type=str, help='Resource provider name. Swagger name=providerName')
+        c.argument('resource_type', type=str, help='Resource type. Swagger name=resourceType')
+        c.argument('resource_name', type=str, help='Resource identifier. Swagger name=resourceName')
+        c.argument('apply_update_name', options_list=['--name', '-n', '--apply-update-name'], type=str,
+                   help='applyUpdate Id. Swagger name=applyUpdateName')
```
Comment (kairu-ms) [inline src/maintenance/azext_maintenance/generated/_params.py:35]: @changlong-liu and @qiaozha Why does help message contain "Swagger name=*"?

Comment (KalpeshChavan12) [reply src/maintenance/azext_maintenance/generated/_params.py:35]: Code is autogenerated from autorest CLI

File: src/maintenance/azext_maintenance/azext_metadata.json (hunk 1 above)
Comment (KalpeshChavan12) [reply src/maintenance/azext_maintenance/azext_metadata.json]: Remove isExperimenatal and isPreview. This is GA release for maintenance sevice.

File: src/maintenance/azext_maintenance/generated/commands.py [hunk 3]
```diff
@@ -52,22 +52,44 @@ def load_command_table(self, _):
...
+        '}',
         client_factory=cf_update)
-    with self.command_group('maintenance update', maintenance_update, client_factory=cf_update,
-                            is_experimental=True) as g:
+    with self.command_group('maintenance update', maintenance_update, client_factory=cf_update) as g:
         g.custom_command('list', 'maintenance_update_list')
         g.custom_command('list-parent', 'maintenance_update_list_parent')
+
+    with self.command_group('maintenance', is_experimental=True):
+        pass
```
Comment (kairu-ms) [inline src/maintenance/azext_maintenance/generated/commands.py:95]: You should not use `is_experimental=True` if it is GA release.

File: src/maintenance/azext_maintenance/generated/_params.py (hunk 2 above)
Comment (KalpeshChavan12) [reply src/maintenance/azext_maintenance/generated/_params.py]: Looks like it due to --debug flag provided to autorest command.
--debug flag should not change behavior of code generation
 
 
https://github.com/Azure/autorest.az/blob/c85ea6a31e4dd7e45b4de0762fcc018b0062f6f6/src/generate/renders/generated/CliParams.ts#L416

Comment (KalpeshChavan12) [reply src/maintenance/azext_maintenance/generated/_params.py]: I have regenerated code without --debug flag.

Please review. PR latest iteration.

File: src/maintenance/azext_maintenance/generated/commands.py (hunk 3 above)
Comment (KalpeshChavan12) [reply src/maintenance/azext_maintenance/generated/commands.py]: I have regenerated code using --extension-mode=stable flag in autorest cmd it removed  above method.

Please review PR

**Comment by KalpeshChavan12** [comment]
@kairu-ms ,

Can you please review PR.

Regards
Kalpesh

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**